feat: add C implementation for `stats/base/dists/uniform/median` (#3970)
stdlib-js/stdlib: lib/node_modules/@stdlib/stats/base/dists/uniform/median/README.md

The [median][median] for a [uniform][uniform-distribution] random variable is

<!-- <equation class="equation" label="eq:uniform_median" align="center" raw="\operatorname{median}\left[ X \right] = 0.5 \cdot ( a + b )" alt="Median for a uniform distribution."> -->

```math
\operatorname{median}\left[ X \right] = 0.5 \cdot ( a + b )
```

<!-- <div class="equation" align="center" data-raw-text="\operatorname{median}\left[ X \right] = 0.5 \cdot ( a + b )" data-equation="eq:uniform_median">
    <img src="..." alt="Median for a uniform distribution.">
    <br>
</div> -->

> 

Reviewer: These changes should be reverted.

<!-- /.examples -->

<!-- C interface documentation. -->

* * *

<section class="c">

## C APIs

<!-- Section to include introductory text. Make sure to keep an empty line after the intro `section` element and another before the `/section` close. -->

<section class="intro">

</section>

<!-- /.intro -->

<!-- C usage documentation. -->

<section class="usage">

### Usage

```c
#include "stdlib/stats/base/dists/uniform/median.h"
```

#### stdlib_base_dists_uniform_median( a, b )

> Returns the [median][median] of a [uniform][uniform-distribution] distribution with minimum value \( a \) and maximum value \( b \).

Reviewer: ```suggestion
Returns the [median][median] of a [uniform][uniform-distribution] distribution with parameters `a` (minimum support) and `b` (maximum support).
```

```c
double out = stdlib_base_dists_uniform_median( 0.0, 1.0 );
// returns 0.5
```

The function accepts the following arguments:

> -   **a**: `[in] double` minimum value.
> -   **b**: `[in] double` maximum value.

Reviewer: ```suggestion
-   **a**: `[in] double` minimum support.
-   **b**: `[in] double` maximum support.
```

```c
double stdlib_base_dists_uniform_median( const double a, const double b );
```

</section>

<!-- /.usage -->

<!-- C API usage notes. Make sure to keep an empty line after the `section` element and another before the `/section` close. -->

<section class="notes">

</section>

<!-- /.notes -->

<!-- C API usage examples. -->

<section class="examples">

### Examples

```c
#include "stdlib/stats/base/dists/uniform/median.h"
#include <stdlib.h>
#include <stdio.h>

> int main( void ) {
>     double a;
>     double b;
>     double y;
>     int i;
> 
>     for ( i = 0; i < 25; i++ ) {
>         a = (double)( rand() % 10 );
>         b = a + (double)( rand() % 10 + 1 ); // ensure b > a
>         y = stdlib_base_dists_uniform_median( a, b );
>         printf( "a: %lf, b: %lf, Median(X;a,b): %lf\n", a, b, y );
>     }
> }

Reviewer: ```suggestion
static double random_uniform( const double min, const double max ) {
    double v = (double)rand() / ( (double)RAND_MAX + 1.0 );
    return min + ( v*(max-min) );
}

int main( void ) {
    double a;
    double b;
    double y;
    int i;

    for ( i = 0; i < 25; i++ ) {
        a = random_uniform( 0.0, 10.0 );
        b = random_uniform( a, 20.0 ); // ensure b > a
        y = stdlib_base_dists_uniform_median( a, b );
        printf( "a: %lf, b: %lf, Median(X;a,b): %lf\n", a, b, y );
    }
}
```
